### ISSUE TITLE
[BFW-5832] [BUG] M73 G-Code status report not correct

### ISSUE THREAD

**Issue by samdurnxn** [prusa3d/Prusa-Firmware-Buddy#4089]
### Printer model

Mini+

### Firmware version

6.0.0, 6.0.3

### Upgrades and modifications

None

### Printing from...

Prusa Link

### Describe the bug

In PrusaSlicer, the option to send progress("Printers -> general -> Supports remaining time") is correctly tick.

For my purpose, I create a python script which read serial data from the printer, every 60s I'd like to know the progress, the remain time and the time until the next change (break or color change). As it's explain in this [article](https://help.prusa3d.com/article/prusa-firmware-specific-g-code-commands_112173), you can get the progress by sending the M73 G-Code without any parameters.

The printer answer this code with : 
M73 Progress: 5%
Time left: 0m
Change: 10m

But the values for time left and change are not updated.

### How to reproduce

You can reproduce this issue like this : 
- connect a device on Linux for example on the serial port of the mini
- open a terminal and run the command 'screen /dev/ttyACM0' 
- start a printing on the mini
- on the terminal, type 'M73' then enter to send the command
- The printer will normally answer with progress informations

### Expected behavior

Normally, the time left value and the time of next change have to be update like the progress value.

**Comment by danopernis** [comment]
Hi @samdurnxn thanks for the report. I confirm the issue is present, we will work on fixing it.

**Comment by samdurnxn** [comment]
just for my information, any news on the fix ?

**Comment by danopernis** [comment]
@samdurnxn the fix was merged into 6.1.0, which was just released for mk4s; release for other printers will follow "in near future" - probably in several weeks.

**Comment by samdurnxn** [comment]
My bad, I check the release summary not the comparaison with previous release.
I will wait for the MK4 version of it.
Thanks for your works!

**Comment by danopernis** [comment]
Nah, it is OK to ask. Nobody expects you to read 1000 commit messages. I would personally prefer smaller releases more often, but that's the way it is :slightly_smiling_face:

**Comment by samdurnxn** [comment]
I fully understand as I'm an developer too ;)

**Comment by danopernis** [comment]
@samdurnxn new version of the firmware 6.1.2 should contain the fix, can you test that?

**Comment by samdurnxn** [comment]
Hello, 
Response of M73 is OK for the time left now but not for the next change.
```
T:231.06/230.00 B:59.98/60.00 X:37.75/36.00 A:50.31/0.00 @:101 B@:67 HBR@:255
> echo: M73 Progress: 1%;
> Time left: 36m;
> Change: 0m;
> ok
> T:230.51/230.00 B:59.93/60.00 X:37.55/36.00 A:50.39/0.00 @:109 B@:64 HBR@:255
> T:230.25/230.00 B:60.06/60.00 X:37.08/36.00 A:50.39/0.00 @:104 B@:20 HBR@:249
> T:230.08/230.00 B:60.05/60.00 X:36.98/36.00 A:50.44/0.00 @:107 B@:40 HBR@:248
> T:229.85/230.00 B:60.06/60.00 X:36.93/36.00 A:50.50/0.00 @:113 B@:27 HBR@:245
.....
> T:229.85/230.00 B:60.00/60.00 X:36.95/36.00 A:50.69/0.00 @:111 B@:50 HBR@:255
> T:229.92/230.00 B:59.98/60.00 X:36.91/36.00 A:50.69/0.00 @:109 B@:53 HBR@:255
> X:122.03 Y:108.91 Z:0.20 E:0.00 Count X:12581 Y:10482 Z:45
> T:230.06/230.00 B:59.94/60.00 X:37.17/36.00 A:50.75/0.00 @:105 B@:57 HBR@:255
> echo:busy: processing
> T:229.68/220.00 B:59.89/55.00 X:37.40/36.00 A:50.80/0.00 @:0 B@:0 HBR@:255
> echo: M73 Progress: 4%;
> Time left: 35m;
> Change: 0m;
```

And on the printer screen, the next change value is "N/A".  Strange because I add little square in PrusaSlicer with many colors changes with the MMU3.
Picture and bgcode attached
[Shape-BoxTestM73.zip](https://github.com/user-attachments/files/16715078/Shape-BoxTestM73.zip)
![screenshot](https://github.com/user-attachments/assets/a6765aa8-02fd-4fba-8aab-80ed4388325b)

**Comment by samdurnxn** [comment]
With a new print, I have to insert break to insert magnets, and the remaining time is working. 
I have the remaining time both on serial link and on screen when I use M601 code. 

Can you confirm that for the moment the remaing time works only with M6xx code ? 
I think this because with the MMU3, the colors changes are done with a Tx code and not a M600.

If I'm right, can you add the Tx code for the remaining time ?

**Comment by samdurnxn** [comment]
I just update the firmware to 6.2 alpha1.
Time left is still OK but the next time change is still at 0m.

Have you plan to handle this value for both M600 and Tx gcode or not ?

I don't know if it's relevant to many people but for me I need this information to add it on a detached monitoring screen.

**Comment by samdurnxn** [comment]
No problem, I understand.
Great if you can add the support for TX gcode.

**Comment by samdurnxn** [comment]
@danopernis ah I understand.

An issue have been already opened on this bug -> https://github.com/prusa3d/PrusaSlicer/issues/13031#issuecomment-2358388811

I had a comment on it.

**Comment by Prusa-Support** [comment]
I seem to understand that part of the remaining issue had to be corrected in Prusa Slicer. Also, I seem to understand that the correction was already made in Prusa Slicer so it may be included with the new [PrusaSlicer 2.9.0-alpha1 Pre-release](https://github.com/prusa3d/PrusaSlicer/releases/tag/version_2.9.0-alpha1).
Please give it a try and consider closing the issue if the fix is confirmed in your tests.

Michele Moramarco
Prusa Research

**Comment by samdurnxn** [comment]
Hello, I download this version and test with only a shape box and some multipainting. If I open the generate GCODE, there is no M73 Cxx line. So for me the bug is not solve yet. I will add a comment in the ticket https://github.com/prusa3d/Prusa-Firmware/issues/4734 too

**Comment by samdurnxn** [comment]
Thanks for the tips.
I done it but any M73 Cxx line have are present in gcode. Maybe I miss something ?

My new config  :
![configTravelerLift](https://github.com/user-attachments/assets/93be0c7c-b157-4ab2-ad82-07c4b0f30927)

I attached the 3mf too like this you can check if there is anything wrong in my setup.
[TestRamping.zip](https://github.com/user-attachments/files/17922700/TestRamping.zip)

**Comment by Prusa-Support** [comment]
@samdurnxn Your file has no user interaction. 

![image](https://github.com/user-attachments/assets/504a2f63-5ceb-41fa-802d-2be7dfc9aff6)

**Try to add a simple manual color change.**

![image](https://github.com/user-attachments/assets/9ffb58ba-d8f9-4b8d-86d7-684261647d11)

Michele Moramarco
Prusa Research

**Comment by samdurnxn** [comment]
I made 4 tests;
- A color change with ramping : M600 and M73 Cx / M73 Dx are present
- A color change without ramping : M600 and M73 Cx / M73 Dx are present
- With Multimaterial paint and with ramping: Tx are present but not M73 Cx /Dx
- With Multimaterial paint and without ramping: Tx are present but not M73 Cx /Dx

I already know that with a manual color change at a specific layer M73 Cx / M73 Dx works but my question is as I have an MMU3, I'd like to know when the next color will be for my monitoring system. I would like to take advantage to color only some parts of a layer with the MMU3.

Actually, tell me if for you, it's normal that the M73 C/D can only be available with M600/M601 code and not with multimaterial painting. If it's normal maybe you can add it in the next feature list of PrusaSlicer ?

**Comment by Prusa-Support** [comment]
> Actually, tell me if for you, it's normal that the M73 C/D can only be available with M600/M601 code and not with multimaterial painting.

I may be just confused but I think the answer is 'yes'.

Please note that MMU is meant for automatic change without user interaction.
This also means thousands of automatic color changes may occur on a simple multi material print, potentially even every few seconds in certain even rather common print scenarios.
We don't plan to provide indications for thousands of automatic filament changes so if this is the ultimate request, I assume the issue may be closed as "not planned"; or it may stay open for some time to collect more requests and other realistic use cases.

Users may plan their manual user interaction in advance, whereas Prusa Slicer will inject M73 codes to help the user plan their manual interaction, supposedly only _when manual user interaction is needed_ and _not for automated long sequences of filament changes_.

Last but not least, given that **this issue is based on Prusa MINI**, it would surely be great to see community projects about it but **MMU functionality** should virtually be identified as **not directly supported**.

Michele Moramarco
Prusa Research

**Comment by samdurnxn** [comment]
Hello, thanks for your explanation. I understand.
